File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        '''
        Toh brute force mein hum sorted property completely ignore kar rahe hain
        is liye ye brute force sirf normal understandind ke liy ehain 
        hence  hm question me used propety ki help leke apna code design karenge
        count=0
        for i in range(len(grid)):
            for  j in range(len(grid[0])):
                if grid[i][j]<0:
                    count+=1
        return count 
        '''
        m = len(grid)
        n = len(grid[0])

        row = m - 1
        col = 0
        count = 0

        while row >= 0 and col < n:

            if grid[row][col] < 0:
                count += n - col
                row -= 1

            else:
                col += 1

        return count

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int m=grid.size();
        int n=grid[0].size();
        int count=0;
        for(auto & row:grid){
            int idx=upper_bound(begin(row),end(row),0,greater<int>())-begin(row);
            count+=(n-idx);
        }
        return count;
    }
};